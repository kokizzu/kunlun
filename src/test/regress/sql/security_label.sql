--
-- Test for facilities of security label
--

-- initial setups
SET client_min_messages TO 'warning';

--DDL_STATEMENT_BEGIN--
DROP ROLE IF EXISTS regress_seclabel_user1;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP ROLE IF EXISTS regress_seclabel_user2;
--DDL_STATEMENT_END--

RESET client_min_messages;

--DDL_STATEMENT_BEGIN--
CREATE USER regress_seclabel_user1 WITH CREATEROLE;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
CREATE USER regress_seclabel_user2;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
CREATE TABLE seclabel_tbl1 (a int, b text);
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
CREATE TABLE seclabel_tbl2 (x int, y text);
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
CREATE VIEW seclabel_view1 AS SELECT * FROM seclabel_tbl2;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
CREATE FUNCTION seclabel_four() RETURNS integer AS $$SELECT 4$$ language sql;
--DDL_STATEMENT_END--

-- ERROR:  Statement 'CREATE DOMAIN' is not supported in Kunlun.
--DDL_STATEMENT_BEGIN--
--CREATE DOMAIN seclabel_domain AS text;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
ALTER TABLE seclabel_tbl1 OWNER TO regress_seclabel_user1;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
ALTER TABLE seclabel_tbl2 OWNER TO regress_seclabel_user2;
--DDL_STATEMENT_END--

--
-- Test of SECURITY LABEL statement without a plugin
--
SECURITY LABEL ON TABLE seclabel_tbl1 IS 'classified';			-- fail
SECURITY LABEL FOR 'dummy' ON TABLE seclabel_tbl1 IS 'classified';		-- fail
SECURITY LABEL ON TABLE seclabel_tbl1 IS '...invalid label...';		-- fail
SECURITY LABEL ON TABLE seclabel_tbl3 IS 'unclassified';			-- fail

SECURITY LABEL ON ROLE regress_seclabel_user1 IS 'classified';			-- fail
SECURITY LABEL FOR 'dummy' ON ROLE regress_seclabel_user1 IS 'classified';		-- fail
SECURITY LABEL ON ROLE regress_seclabel_user1 IS '...invalid label...';		-- fail
SECURITY LABEL ON ROLE regress_seclabel_user3 IS 'unclassified';			-- fail

-- clean up objects
--DDL_STATEMENT_BEGIN--
DROP FUNCTION seclabel_four();
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP DOMAIN seclabel_domain;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP VIEW seclabel_view1;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP TABLE seclabel_tbl1;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP TABLE seclabel_tbl2;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP USER regress_seclabel_user1;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP USER regress_seclabel_user2;
--DDL_STATEMENT_END--