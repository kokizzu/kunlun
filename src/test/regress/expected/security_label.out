--
-- Test for facilities of security label
--
-- initial setups
SET client_min_messages TO 'warning';
SET
DROP ROLE IF EXISTS regress_seclabel_user1;
DROP ROLE
DROP ROLE IF EXISTS regress_seclabel_user2;
DROP ROLE
RESET client_min_messages;
RESET
CREATE USER regress_seclabel_user1 WITH CREATEROLE;
CREATE ROLE
CREATE USER regress_seclabel_user2;
CREATE ROLE
CREATE TABLE seclabel_tbl1 (a int, b text);
CREATE TABLE
CREATE TABLE seclabel_tbl2 (x int, y text);
CREATE TABLE
CREATE VIEW seclabel_view1 AS SELECT * FROM seclabel_tbl2;
CREATE VIEW
CREATE FUNCTION seclabel_four() RETURNS integer AS $$SELECT 4$$ language sql;
CREATE FUNCTION
CREATE DOMAIN seclabel_domain AS text;
psql:sql/security_label.sql:20: ERROR:  Statement 'CREATE DOMAIN' is not supported in Kunlun.
ALTER TABLE seclabel_tbl1 OWNER TO regress_seclabel_user1;
ALTER TABLE
ALTER TABLE seclabel_tbl2 OWNER TO regress_seclabel_user2;
ALTER TABLE
--
-- Test of SECURITY LABEL statement without a plugin
--
SECURITY LABEL ON TABLE seclabel_tbl1 IS 'classified';			-- fail
psql:sql/security_label.sql:28: ERROR:  no security label providers have been loaded
SECURITY LABEL FOR 'dummy' ON TABLE seclabel_tbl1 IS 'classified';		-- fail
psql:sql/security_label.sql:29: ERROR:  security label provider "dummy" is not loaded
SECURITY LABEL ON TABLE seclabel_tbl1 IS '...invalid label...';		-- fail
psql:sql/security_label.sql:30: ERROR:  no security label providers have been loaded
SECURITY LABEL ON TABLE seclabel_tbl3 IS 'unclassified';			-- fail
psql:sql/security_label.sql:31: ERROR:  no security label providers have been loaded
SECURITY LABEL ON ROLE regress_seclabel_user1 IS 'classified';			-- fail
psql:sql/security_label.sql:33: ERROR:  no security label providers have been loaded
SECURITY LABEL FOR 'dummy' ON ROLE regress_seclabel_user1 IS 'classified';		-- fail
psql:sql/security_label.sql:34: ERROR:  security label provider "dummy" is not loaded
SECURITY LABEL ON ROLE regress_seclabel_user1 IS '...invalid label...';		-- fail
psql:sql/security_label.sql:35: ERROR:  no security label providers have been loaded
SECURITY LABEL ON ROLE regress_seclabel_user3 IS 'unclassified';			-- fail
psql:sql/security_label.sql:36: ERROR:  no security label providers have been loaded
-- clean up objects
DROP FUNCTION seclabel_four();
DROP FUNCTION
DROP DOMAIN seclabel_domain;
psql:sql/security_label.sql:40: ERROR:  type "seclabel_domain" does not exist
DROP VIEW seclabel_view1;
DROP VIEW
DROP TABLE seclabel_tbl1;
DROP TABLE
DROP TABLE seclabel_tbl2;
DROP TABLE
DROP USER regress_seclabel_user1;
DROP ROLE
DROP USER regress_seclabel_user2;
DROP ROLE
